Fix BangOlufsenSource enum member names

mj23000 opened the pull request:
The commit: [Fix enum comparisons and return values
](https://github.com/home-assistant/core/commit/0ea91515cf5245187d6276d50f85f09ab71e19d2) improved naming for several enums in the bang_olufsen integration, but also modified the names of the BangOlufsenSource enum members, which used to correspond to source ids used with the API. 

The commit broke a few source checks, so this PR is a simple revert for those member names in the BangOlufsenSource enum and clarifications where the enum is used.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

mj23000 committed:
Revert BangOlufsenSource ids
Fix enum comparisons and return values

mj23000 commented on homeassistant/components/bang_olufsen/media_player.py:
```diff
@@ -336,8 +336,8 @@ async def _update_source_change(self, data: Source) -> None:
-            BangOlufsenSource.LINE_IN,
-            BangOlufsenSource.SPDIF,
+            BangOlufsenSource.lineIn.name,
+            BangOlufsenSource.spdif.name,
```
Each B&O source has an id and a friendly name, which is what the Enum was. For example this code here would fail because now the self._source_change.id would never be equal to "LINE_IN" or "Line-In" because the real source id is "lineIn", so the playback progress was never reset.

joostlek replied:
Hmm, I am interested if we can also use translations for the state attributes as in a climate entity (for the presets). This way we could keep the friendly names out of the code. Enums should be full capital with underscores. Another option would be to make a map to translate the enum to friendly names. Sounds like a lot of duplication, but it makes it a lot cleaner.

mj23000 replied:
Ok. I've converted the enum class to just be a class with members. I think this retains the usability and readability of the enum for this use case.

mj23000 committed:
Convert enum class to class

joostlek commented on homeassistant/components/bang_olufsen/const.py:
```diff
@@ -10,27 +10,27 @@
-    DEEZER = "Deezer"
-    BEOLINK = "Networklink"
-    TIDAL_CONNECT = "Tidal Connect"
+class BangOlufsenSource:
```
This can still be an enum right?

mj23000 replied:
It can, but using the enum is a bit more confusing with having to use the "name" and "value" members

mj23000 replied:
Yeah, it's a bit messy. Honestly I think that the class should be removed altogether, most of its uses are at best mildly useful. I think I'll try to retain some of the functionality and remove it.

joostlek replied:
Like, I think we can improve this more than we do now. If we can move parts to the lib, that's always great

mj23000 replied:
A future API update will remove some of the need for BangOlufsenSource, but I think the current state (of this PR) should be fine for now

joostlek commented on homeassistant/components/bang_olufsen/const.py:
```diff
@@ -10,27 +10,27 @@
+class BangOlufsenSource:
+    """Class used for associating device source ids with friendly names. May not include all sources."""
+
+    uri_streamer = Source(name="Audio Streamer", id="uriStreamer")
+    bluetooth = Source(name="Bluetooth", id="bluetooth")
+    air_play = Source(name="AirPlay", id="airPlay")
+    chromecast = Source(name="Chromecast built-in", id="chromeCast")
+    spotify = Source(name="Spotify Connect", id="spotify")
+    generator = Source(name="Tone Generator", id="generator")
+    line_in = Source(name="Line-In", id="lineIn")
+    spdif = Source(name="Optical", id="spdif")
+    net_radio = Source(name="B&O Radio", id="netRadio")
+    local = Source(name="Local", id="local")
+    dlna = Source(name="DLNA", id="dlna")
+    qplay = Source(name="QPlay", id="qplay")
+    wpl = Source(name="Wireless Powerlink", id="wpl")
+    pl = Source(name="Powerlink", id="pl")
+    tv = Source(name="TV", id="tv")
+    deezer = Source(name="Deezer", id="deezer")
+    beolink = Source(name="Networklink", id="beolink")
+    tidal_connect = Source(name="Tidal Connect", id="tidalConnect")
```
Let's keep it uppercase

mj23000 replied:
I've made the members uppercase

mj23000 committed:
Capitalize BangOlufsenSource static members
Add Typing to BangOlufsenSource members
Use id instead of name for BangOlufsenSource where applicable
Remove unused BangOlufsenSource members

mj23000 committed:
Merge branch 'dev' into bang_olufsen_refix_enum

mj23000 committed:
Merge branch 'dev' into bang_olufsen_refix_enum

frenck reviewed (approved):
Thanks, @mj23000 👍 

../Frenck
